### PR TITLE
CH4/carbon simulation updates: (1) Use 0.1 degree mask files; (2) Prevent reading the same variable from disk more than once

### LINKED ISSUE #2268: 0.001 x 0.001 degree masks cause performance slowdown in GCHP for carbon simulations
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://geos-chem.readthedocs.io/en/stable/help-and-reference/CONTRIBUTING.html)

### Overview
In the GCHP carbon simulation, the `Scarpelli_Canada` and `Scarpelli_Mexico` emissions use these mask files described below..  @lizziel found that reading/regridding these mask files causes a substantial performance penalty in GCHP.  This may also cause a slowdown in GCClassic CH4 and carbon simulations (including inversions for the IMI).

**Specification in HEMCO_Config.rc:**
```console
1001 MEX_MASK          $ROOT/MASKS/v2024-01/Mexico_Mask.001x001.nc        MASK 2000/1/1/0 C xy 1 1 -118/17/-95/33
1010 MEX_MASK_MIRROR   $ROOT/MASKS/v2024-01/Mexico_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -118/17/-95/33
1002 CAN_MASK          $ROOT/MASKS/v2024-01/Canada_Mask.001x001.nc        MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-01/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
```
**Specification in ExtData.rc**
```console
MEX_MASK          1 N Y - none none MASK ./HcoDir/MASKS/v2024-01/Mexico_Mask.001x001.nc
MEX_MASK_MIRROR   1 N Y - none none MASK ./HcoDir/MASKS/v2024-01/Mexico_Mask_Mirror.001x001.nc
CAN_MASK          1 N Y - none none MASK ./HcoDir/MASKS/v2024-01/Canada_Mask.001x001.nc
CAN_MASK_MIRROR   1 N Y - none none MASK ./HcoDir/MASKS/v2024-01/Canada_Mask_Mirror.001x001.nc
```

**Mask files**
The mask files are at high horizontal resolution (0.001 x 0.001 degree) and include a subsetted area around the region of interest (Canada or Mexico).  The mask coverages are shown below.  NOTE: The mirror masks images are omitted, they are just the inverse of the original masks (i.e. NOT Mexico and NOT Canada).

2024_04_25_17_26_28_Canada_Mask 001x001 nc_Ubuntu_

2024_04_25_17_29_17_Mexico_Mask 001x001 nc_Ubuntu_

**Proposed solution**
Creating smaller mask files will reduce the computational overhead involved in reading/regridding.  Both the `Scarpelli_Canada` and `Scarpelli_Mexico` emissions are 0.1 x 0.1 degrees, so we shouldn't need mask files with resolution finer than 0.1 x 0.1 resolution.

**Related issues**
- https://github.com/geoschem/GCHP/issues/411

Tagging @msulprizio @lizziel @laestrada

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://geos-chem.readthedocs.io/en/stable/help-and-reference/CONTRIBUTING.html)

### Describe the update
This PR does the following:

1. Replaces the existing 0.01 x 0.01 degree mask files for the Scarpelli_Canada, Scarpelli_Mexico, and GHGI CH4 emissions with 0.1 x 0.1 degree mask files.  Using the higher-resolution mask files was causing a I/O bottleneck in GCHP due to the large number of grid points in the file.

2. Update the `HEMCO_Config.rc`  files so that the `*COAST*` entries are piggybacked onto the preceding entry.  In other words, change this:
```console
(((Scarpelli_Canada
0 CAN_OIL_GAS_COMBUSTION  $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_combustion_2018.nc  oil_gas_combustion_total  2018/1/1/0 C xy kg/m2/s CH4 1002 1/2 100
0 CAN_OIL_GAS_LEAKAGE     $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_leakage_2018.nc     oil_gas_leakage_total     2018/1/1/0 C xy kg/m2/s CH4 1002 1/2 100
0 CAN_OIL_GAS_VENT_FLARE  $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_vent_flare_2018.nc  oil_gas_vent_flare_total  2018/1/1/0 C xy kg/m2/s CH4 1002 1/2 100
0 CAN_COAL                $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_coal_2018.nc                coal_total                2018/1/1/0 C xy kg/m2/s CH4 1002 3   100
0 CAN_LIVESTOCK           $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_livestock_2018.nc           livestock_total           2018/1/1/0 C xy kg/m2/s CH4 1002 4   100
0 CAN_SOLID_WASTE         $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_solid_waste_2018.nc         solid_waste_total         2018/1/1/0 C xy kg/m2/s CH4 1002 5   100
0 CAN_WASTEWATER          $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_wastewater_2018.nc          wastewater_total          2018/1/1/0 C xy kg/m2/s CH4 1002 6   100
0 CAN_OTHER               $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_other_minor_sources_2018.nc other_minor_sources_total 2018/1/1/0 C xy kg/m2/s CH4 1002 8   100

### Make sure to include offshore/coastal emissions (Hier=1 to add to EDGAR, Hier=5 to add to GFEI) ###
0 CAN_OIL_GAS_COMBUSTION_COAST  $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_combustion_2018.nc  oil_gas_combustion_total  2018/1/1/0 C xy kg/m2/s CH4 1011 1/2 5
0 CAN_OIL_GAS_LEAKAGE_COAST     $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_leakage_2018.nc     oil_gas_leakage_total     2018/1/1/0 C xy kg/m2/s CH4 1011 1/2 5
0 CAN_OIL_GAS_VENT_FLARE_COAST  $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_vent_flare_2018.nc  oil_gas_vent_flare_total  2018/1/1/0 C xy kg/m2/s CH4 1011 1/2 5
0 CAN_COAL_COAST                $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_coal_2018.nc                coal_total                2018/1/1/0 C xy kg/m2/s CH4 1011 3   5
0 CAN_LIVESTOCK_COAST           $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_livestock_2018.nc           livestock_total           2018/1/1/0 C xy kg/m2/s CH4 1011 4   1
0 CAN_SOLID_WASTE_COAST         $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_solid_waste_2018.nc         solid_waste_total         2018/1/1/0 C xy kg/m2/s CH4 1011 5   1
0 CAN_WASTEWATER_COAST          $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_wastewater_2018.nc          wastewater_total          2018/1/1/0 C xy kg/m2/s CH4 1011 6   1
0 CAN_OTHER_COAST               $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_other_minor_sources_2018.nc other_minor_sources_total 2018/1/1/0 C xy kg/m2/s CH4 1011 8   1
)))Scarpelli_Canada
```
to this
```console
(((Scarpelli_Canada
0 CAN_OIL_GAS_COMBUSTION        $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_combustion_2018.nc  oil_gas_combustion_total  2018/1/1/0 C xy kg/m2/s CH4 1002 1/2 100
0 CAN_OIL_GAS_COMBUSTION_COAST  -                                                                        -                         -          - -  -       CH4 1011 1/2 5
0 CAN_OIL_GAS_LEAKAGE           $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_leakage_2018.nc     oil_gas_leakage_total     2018/1/1/0 C xy kg/m2/s CH4 1002 1/2 100
0 CAN_OIL_GAS_LEAKAGE_COAST     -                                                                        -                         -          - -  -       CH4 1011 1/2 5
0 CAN_OIL_GAS_VENT_FLARE        $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_oil_gas_vent_flare_2018.nc  oil_gas_vent_flare_total  2018/1/1/0 C xy kg/m2/s CH4 1002 1/2 100
0 CAN_OIL_GAS_VENT_FLARE_COAST  -                                                                        -                         -          - -  -       CH4 1011 1/2 5
0 CAN_COAL                      $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_coal_2018.nc                coal_total                2018/1/1/0 C xy kg/m2/s CH4 1002 3   100
0 CAN_COAL_COAST                -                                                                        -                         -          - -  -       CH4 1011 3   5
0 CAN_LIVESTOCK                 $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_livestock_2018.nc           livestock_total           2018/1/1/0 C xy kg/m2/s CH4 1002 4   100
0 CAN_LIVESTOCK_COAST           -                                                                        -                         -          - -  -       CH4 1011 4   1
0 CAN_SOLID_WASTE               $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_solid_waste_2018.nc         solid_waste_total         2018/1/1/0 C xy kg/m2/s CH4 1002 5   100
0 CAN_SOLID_WASTE_COAST         -                                                                        -                         -          - -  -       CH4 1011 5   1
0 CAN_WASTEWATER                $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_wastewater_2018.nc          wastewater_total          2018/1/1/0 C xy kg/m2/s CH4 1002 6   100
0 CAN_WASTEWATER_COAST          -                                                                        -                         -          - -  -       CH4 1011 6   1
0 CAN_OTHER                     $ROOT/CH4/v2022-01/Scarpelli_Canada/can_emis_other_minor_sources_2018.nc other_minor_sources_total 2018/1/1/0 C xy kg/m2/s CH4 1002 8   100
0 CAN_OTHER_COAST               -                                                                        -                         -          - -  -       CH4 1011 8   1
```
etc. for other relevant inventories.  This will prevent having to read the same variable from each file twice, and should result in a performance benefit

3. New mask files will be placed in `ExtData/HEMCO/v2024-04/`, and will soon be available on the s3://gcgrid and WashU data portal.

### Related Github Issue(s)

- See https://github.com/geoschem/GCHP/issues/411
- Closes #2268 

Tagging @kbowman77 @hannahnesser

### CODE REVIEW

File: run/GCHP/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon [hunk 1]
```diff
@@ -1682,18 +1683,18 @@ ${RUNDIR_CO2_COPROD}
...
-1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-01/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
+1002 CAN_MASK          $ROOT/MASKS/v2024-04/Canada_Mask.001x001.nc        MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
+1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-04/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
 )))Scarpelli_Canada
 
 (((GHGI_v2.or.GHGI_v2_Express_Ext
-1008 CONUS_MASK        $ROOT/MASKS/v2018-09/CONUS_Mask.001x001.nc         MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
-1009 CONUS_MASK_MIRROR $ROOT/MASKS/v2018-09/CONUS_Mask_Mirror.001x001.nc  MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
+1008 CONUS_MASK        $ROOT/MASKS/v2024-04/CONUS_Mask.001x001.nc         MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
+1009 CONUS_MASK_MIRROR $ROOT/MASKS/v2024-04/CONUS_Mask_Mirror.001x001.nc  MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
```
Comment (msulprizio) [inline run/GCHP/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon:1697]: I think the filenames here should use 01x01 and not 001x001.

File: run/GEOS/GEOSCHEMchem_GridComp.rc [hunk 2]
```diff
@@ -226,7 +226,7 @@ Bry_Cly_Family_Transport: 1
 # CO production from CO2 photolysis
 #
 CO_production_from_CO2_photolysis: 0
-CO2photolysisFile: /gpfsm/dnb52/projects/p10/dasilva/fvInput/ExtData/chemistry/GEOSCHEMchem/v0.0.0/CHEM_INPUTS/StratChem/SC.J_20_12_79_72_200_45.jpl15.nc4
+CO2photolysisFile: /discover/nobackup/projects/gmao/share/dasilva/fvInput/ExtData/chemistry/GEOSCHEMchem/v0.0.0/CHEM_INPUTS/StratChem/SC.J_20_12_79_72_200_45.jpl15.nc4
```
Comment (msulprizio) [inline run/GEOS/GEOSCHEMchem_GridComp.rc:229]: Why are the changes in this file here? Are they reverting changes made by @lizziel in PR https://github.com/geoschem/geos-chem/pull/2249?

File: run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.CH4 [hunk 3]
```diff
@@ -935,18 +935,18 @@ ${RUNDIR_GLOBAL_Cl}
...
-1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-01/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
+1002 CAN_MASK          $ROOT/MASKS/v2024-04/Canada_Mask.001x001.nc        MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
+1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-04/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
 )))Scarpelli_Canada
 
 (((GHGI_v2.or.GHGI_v2_Express_Ext
-1008 CONUS_MASK        $ROOT/MASKS/v2018-09/CONUS_Mask.001x001.nc         MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
-1009 CONUS_MASK_MIRROR $ROOT/MASKS/v2018-09/CONUS_Mask_Mirror.001x001.nc  MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
+1008 CONUS_MASK        $ROOT/MASKS/v2024-04/CONUS_Mask.001x001.nc         MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
+1009 CONUS_MASK_MIRROR $ROOT/MASKS/v2024-04/CONUS_Mask_Mirror.001x001.nc  MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
```
Comment (msulprizio) [inline run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.CH4:949]: I think the filenames here should use 01x01 and not 001x001.

File: run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon [hunk 4]
```diff
@@ -1682,18 +1683,18 @@ ${RUNDIR_CO2_COPROD}
...
-1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-01/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
+1002 CAN_MASK          $ROOT/MASKS/v2024-04/Canada_Mask.001x001.nc        MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
+1011 CAN_MASK_MIRROR   $ROOT/MASKS/v2024-04/Canada_Mask_Mirror.001x001.nc MASK 2000/1/1/0 C xy 1 1 -141/40/-52/85
 )))Scarpelli_Canada
 
 (((GHGI_v2.or.GHGI_v2_Express_Ext
-1008 CONUS_MASK        $ROOT/MASKS/v2018-09/CONUS_Mask.001x001.nc         MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
-1009 CONUS_MASK_MIRROR $ROOT/MASKS/v2018-09/CONUS_Mask_Mirror.001x001.nc  MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
+1008 CONUS_MASK        $ROOT/MASKS/v2024-04/CONUS_Mask.001x001.nc         MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
+1009 CONUS_MASK_MIRROR $ROOT/MASKS/v2024-04/CONUS_Mask_Mirror.001x001.nc  MASK 2000/1/1/0 C xy 1 1 -140/20/-50/60
```
Comment (msulprizio) [inline run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon:1697]: I think the filenames here should use 01x01 and not 001x001.

**Review by msulprizio — CHANGES_REQUESTED** [review]
The filenames in several HEMCO_Config.rc files need to be changed to use resolution string 01x01 instead of 001x001 for consistency with the changes in ExtData.rc.carbon.

Also, changes are made to several GEOS input files. Are those intended? Are they reverting changes made in https://github.com/geoschem/geos-chem/pull/2249?

File: run/GEOS/GEOSCHEMchem_GridComp.rc (hunk 2 above)
Comment (yantosca) [reply run/GEOS/GEOSCHEMchem_GridComp.rc:229]: I plan to do integration test for this PR and also PR #2271.  I just haven't pushed that yet while the HETP (alpha.15) benchmarks are running).

File: run/GCHP/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon (hunk 1 above)
Comment (yantosca) [reply run/GCHP/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon]: Now fixed in 1e4465d

File: run/GEOS/GEOSCHEMchem_GridComp.rc (hunk 2 above)
Comment (yantosca) [reply run/GEOS/GEOSCHEMchem_GridComp.rc]: PR #2271 is now merged.

File: run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.CH4 (hunk 3 above)
Comment (yantosca) [reply run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.CH4]: Now fixed in https://github.com/geoschem/geos-chem/commit/1e4465dec553e81cbfd45ce6406fb379ba7cc2aa

File: run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon (hunk 4 above)
Comment (yantosca) [reply run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon]: Now fixed in https://github.com/geoschem/geos-chem/commit/1e4465dec553e81cbfd45ce6406fb379ba7cc2aa

**Comment by yantosca** [comment]
@msulprizio: I fixed the remaining issues in the config files and am rerunning integration tests.  Stay tuned for results.

**Comment by yantosca** [comment]
@msulprizio: I will have a better idea once the integration tests finish.  I can look at the timing information w/r/t the previous integration test.

**Comment by yantosca** [comment]
I've fixed some issues in the GHGI_EE* entries in `ExtData.rc.carbon` and have merged into 14.4.0 (locally) so that I can run integration tests. Stay tuned for results.

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #041c5d6 GEOS-Chem submod update: Merge PR #2281 (Change input file paths for GEOS)
GEOS-Chem #7dad2686f Merge PR #2272 (CH4/carbon simulation updates)
HEMCO     #fddcae5 Update version to 3.8.1 for release

Using 24 OpenMP threads
Number of execution tests: 29

Submitted as SLURM job: 30237963
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
Due to cluster issues, I ran a "quick" set of GCHP integration tests (as this PR only affects the CH4/carbon simulations).  All tests passed.
```console
==============================================================================
GCHP: Execution Test Results

GCHP      #dfe6a3e GEOS-Chem submod update: Merge PR #2281 (Change input file paths for GEOS)
GEOS-Chem #7102094f5 Merge PR #2272 (CH4/carbon simulation updates)
HEMCO     #

Number of execution tests: 7

Submitted as SLURM job: 30324185
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_merra2_carbon..................................Execute Simulation....PASS
gchp_merra2_carbon_CH4..............................Execute Simulation....PASS
gchp_merra2_carbon_CO...............................Execute Simulation....PASS
gchp_merra2_carbon_CO2..............................Execute Simulation....PASS
gchp_merra2_carbon_OCS..............................Execute Simulation....PASS
gchp_merra2_tagO3...................................Execute Simulation....PASS
gchp_merra2_TransportTracers........................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 7
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Pull request merged**